rpc/src/v1/helpers/dispatch.rs

```diff
@@ -129,31 +134,15 @@ impl<C: MiningBlockChainClient, M: MinerService> Dispatcher for FullDispatcher<C
 				data: filled.data,
 			};
 
-			let hash = t.hash(network_id);
 			if accounts.is_hardware_address(address) {
-				let mut stream = rlp::RlpStream::new();
-				t.rlp_append_unsigned_transaction(&mut stream, network_id);
-				let signature = try_bf!(
-					accounts.sign_with_hardware(address, &stream.as_raw())
-						.map_err(|e| {
-							debug!(target: "miner", "Error signing transaction with hardware wallet: {}", e);
-							errors::account("Error signing transaction with hardware wallet", e)
-						})
-				);
-				let signed = try_bf!(
-					SignedTransaction::new(t.with_signature(signature, network_id))
-						.map_err(|e| {
-						  debug!(target: "miner", "Hardware wallet has produced invalid signature: {}", e);
-						  errors::account("Invalid signature generated", e)
-						})
-				);
-				WithToken::No(signed)
+				hardware_signature(&*accounts, address, t, network_id).map(WithToken::No)
```

Comment:
Just factoring out common code shared between the `FullDispatcher` and the `LightDispatcher`. The dispatcher's role is to fill optional fields of transaction requests, fulfill signer requests, and dispatch transactions onto the network.